RPT_Application/src/main/java/com/harman/web_pages/EmployeesCRUDView.java

```diff
@@ -0,0 +1,231 @@
+package com.harman.web_pages;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import com.harman.report_database.ReportService;
+import com.harman.user_database.Authority;
+import com.harman.user_database.AuthorityType;
+import com.harman.user_database.Employee;
+import com.harman.user_database.EmployeeService;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.dialog.Dialog;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.grid.GridVariant;
+import com.vaadin.flow.component.grid.contextmenu.GridContextMenu;
+import com.vaadin.flow.component.html.Label;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.spring.annotation.UIScope;
+
+@Route("users")
+@UIScope
+public class EmployeesCRUDView extends VerticalLayout {
+
+	private static final long serialVersionUID = 2435396446114365838L;
+	
+	@Autowired
+	private EmployeeService employeeService;
+	
+	@Autowired
+	private ReportService reportService;
+
+	@Autowired
+	private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+	public EmployeesCRUDView(EmployeeService employeeService) {
+
+		Grid<Employee> grid = new Grid<>(Employee.class);
+		grid.setItems(employeeService.findAll());
+		grid.setColumns("id", "username", "authority");
+
+		grid.getColumnByKey("id").setFlexGrow(0);
+
+		GridContextMenu<Employee> contextMenu = new GridContextMenu<>(grid);
+
+		contextMenu.addItem("Remove", e -> {
+			e.getItem().ifPresent(a -> {
+				createAreYouSureDialog(grid, a);
+			});
+		});
+
+		contextMenu.addItem("Edit", e -> {
+			e.getItem().ifPresent(a -> {
+				createEditDialog(grid, a);
+			});
+		});
+
+		grid.addThemeVariants(GridVariant.LUMO_ROW_STRIPES, GridVariant.LUMO_COLUMN_BORDERS);
+
+		Button addButton = new Button("Add Employee");
+		addButton.addClickListener(e -> createSaveDialog(grid));
+		
+		grid.addItemDoubleClickListener(e -> {
+			createEmployeeReportsDialog(e.getItem());
+		});
+		
+		setSizeFull();
+		add(addButton, grid);
+	}
+
+	private void createEmployeeReportsDialog(Employee employee) {
```

Comment:
We have several dialogs in different places in the code. Maybe we can create the class and configure it accordingly.

Reply:
Till now I've tried to take out those methods from their classes but there are some dependency loops, it's not easy for me to find a way how to move them.
Now we have new DialogCreatorMaster class that has those methods that it's clear  for me that they will work there. 
When all UI will be done I'll try to put all of them there,  maybe some overwriting?